src/setup/helpers.ts

```diff
@@ -107,3 +104,49 @@ export function hashSignedObj(obj): string {
   }
   return crypto.hashObj(obj, true)
 }
+
+/**
+@param rawPayload: any - The original payload stripped of the signatures
+@param sigs: Sign[] - The signatures to verify
+@param allowedPubkeys: {[pubkey: string]: DevSecurityLevel} - The public keys that are allowed to sign the payload
+@param minSigRequired: number - The minimum number of signatures required
+@param requiredSecurityLevel: DevSecurityLevel - The minimum security level required to sign the payload
+@returns boolean - True if the payload is signed by the required number of authorized public keys with the required security level
+**/
+export function verifyMultiSigs(
+  rawPayload: object,
+  sigs: Sign[],
+  allowedPubkeys: { [pubkey: string]: DevSecurityLevel },
+  minSigRequired: number,
+  requiredSecurityLevel: DevSecurityLevel
+): boolean {
+  if (sigs.length < minSigRequired) return false
+
+  // no reason to allow more signatures than allowedPubkeys exist
+  // this also prevent loop exhaustion
+  if (sigs.length > Object.keys(allowedPubkeys).length) return false
+
+  let validSigs = 0
+  const payload_hash = ethers.keccak256(ethers.toUtf8Bytes(Utils.safeStringify(rawPayload)))
+  const seen = new Set()
+
+  for (let i = 0; i < sigs.length; i++) {
+    // The sig has already been seen
```

Comment:
very minor comment:   (should be "has not already been seen")

Reply:
Updated